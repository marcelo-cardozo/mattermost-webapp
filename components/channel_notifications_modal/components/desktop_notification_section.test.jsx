// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {NotificationLevels} from 'utils/constants';

import DesktopNotificationSection from 'components/channel_notifications_modal/components/desktop_notification_section.jsx';
import * as Utils from '../../../utils/utils';

describe('components/channel_notifications_modal/DesktopNotificationSection', () => {
    const baseProps = {
        onChange: () => {}, //eslint-disable-line no-empty-function
        onSubmit: () => {}, //eslint-disable-line no-empty-function
        serverError: '',
        onCollapseSection: () => {}, //eslint-disable-line no-empty-function
        memberNotifyLevel: NotificationLevels.DEFAULT,
        globalNotifyLevel: NotificationLevels.DEFAULT,
        memberDesktopSound: undefined,
        globalDesktopSound: 'true',
        memberDesktopNotificationSound: undefined,
        globalDesktopNotificationSound: 'Bing',
        handleUpdateDesktopSound: () => {}, //eslint-disable-line no-empty-function
        handleUpdateDesktopNotificationSound: () => {}, //eslint-disable-line no-empty-function
    };

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('should match snapshot default global settings', () => {
        const wrapper = shallow(
            <DesktopNotificationSection {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should show notification options', () => {
        const userAgentMock = jest.requireMock('../../../utils/user_agent');
        userAgentMock.isDesktopApp.mockImplementationOnce(() => false);

        jest.spyOn(Utils, 'hasSoundOptions').mockImplementationOnce(() => true);

        const props = {
            ...baseProps,
            globalNotifyLevel: NotificationLevels.ALL,
            globalDesktopSound: 'true',
        };

        const wrapper = shallow(
            <DesktopNotificationSection {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should not show notification options when browser not supported', () => {
        const userAgentMock = jest.requireMock('../../../utils/user_agent');
        userAgentMock.isDesktopApp.mockImplementationOnce(() => true);

        jest.spyOn(Utils, 'hasSoundOptions').mockImplementationOnce(() => false);

        const props = {
            ...baseProps,
            globalNotifyLevel: NotificationLevels.ALL,
            globalDesktopSound: 'true',
        };

        const wrapper = shallow(
            <DesktopNotificationSection {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should not show notification section when notify level is none', () => {
        const props = {
            ...baseProps,
            globalNotifyLevel: NotificationLevels.NONE,
        };

        const wrapper = shallow(
            <DesktopNotificationSection {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should show reset to defaults message', () => {
        jest.spyOn(Utils, 'hasSoundOptions').mockImplementationOnce(() => true);
        jest.spyOn(Utils, 'isMobile').mockImplementationOnce(() => false);

        const props = {
            ...baseProps,
            globalNotifyLevel: NotificationLevels.NONE,
            memberNotifyLevel: NotificationLevels.MENTION,
        };

        const wrapper = shallow(
            <DesktopNotificationSection {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should show reset to defaults icon for mobile', () => {
        jest.spyOn(Utils, 'hasSoundOptions').mockImplementationOnce(() => true);
        jest.spyOn(Utils, 'isMobile').mockImplementationOnce(() => true);

        const props = {
            ...baseProps,
            globalNotifyLevel: NotificationLevels.NONE,
            memberNotifyLevel: NotificationLevels.MENTION,
        };

        const wrapper = shallow(
            <DesktopNotificationSection {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
