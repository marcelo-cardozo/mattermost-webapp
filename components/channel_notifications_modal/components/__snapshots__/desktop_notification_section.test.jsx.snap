// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_notifications_modal/DesktopNotificationSection should match snapshot default global settings 1`] = `
<SettingItemMax
  actionButton={null}
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`components/channel_notifications_modal/DesktopNotificationSection should not show notification options when browser not supported 1`] = `
<SettingItemMax
  actionButton={null}
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={true}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <FormattedMessage
                defaultMessage="Notification sound"
                id="channel.user.settings.notifications.desktop.sound"
              />
            </legend>
            <br />
            <FormattedMessage
              defaultMessage="Please configure notification sounds in your browser settings"
              id="channel.user.settings.notifications.soundConfig"
            />
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`components/channel_notifications_modal/DesktopNotificationSection should not show notification section when notify level is none 1`] = `
<SettingItemMax
  actionButton={null}
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={true}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`components/channel_notifications_modal/DesktopNotificationSection should show notification options 1`] = `
<SettingItemMax
  actionButton={null}
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={true}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <FormattedMessage
                defaultMessage="Notification sound"
                id="channel.user.settings.notifications.desktop.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={true}
                  id="soundOn"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="On"
                  id="channel.user.settings.notifications.on"
                />
              </label>
              <br />
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="soundOff"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="Off"
                  id="channel.user.settings.notifications.off"
                />
              </label>
              <br />
            </div>
            <div
              className="pt-2"
            >
              <StateManager
                className="react-select notification-sound-dropdown"
                classNamePrefix="react-select"
                clearable={false}
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                id="displaySoundNotification"
                isSearchable={false}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Bing",
                      "value": "Bing",
                    },
                    Object {
                      "label": "Crackle",
                      "value": "Crackle",
                    },
                    Object {
                      "label": "Down",
                      "value": "Down",
                    },
                    Object {
                      "label": "Hello",
                      "value": "Hello",
                    },
                    Object {
                      "label": "Ripple",
                      "value": "Ripple",
                    },
                    Object {
                      "label": "Upstairs",
                      "value": "Upstairs",
                    },
                  ]
                }
                value={
                  Object {
                    "label": "Bing",
                    "value": "Bing",
                  }
                }
              />
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Mattermost Desktop Apps."
                id="channel.user.settings.notifications.sounds_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`components/channel_notifications_modal/DesktopNotificationSection should show reset to defaults icon for mobile 1`] = `
<SettingItemMax
  actionButton={
    <button
      aria-labelledby="desktopTitle desktopAction"
      className="color--link cursor--pointer style--none"
      id="desktopAction"
      onClick={[Function]}
    >
      <RepeatIcon />
    </button>
  }
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={true}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <FormattedMessage
                defaultMessage="Notification sound"
                id="channel.user.settings.notifications.desktop.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={true}
                  id="soundOn"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="On"
                  id="channel.user.settings.notifications.on"
                />
              </label>
              <br />
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="soundOff"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="Off"
                  id="channel.user.settings.notifications.off"
                />
              </label>
              <br />
            </div>
            <div
              className="pt-2"
            >
              <StateManager
                className="react-select notification-sound-dropdown"
                classNamePrefix="react-select"
                clearable={false}
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                id="displaySoundNotification"
                isSearchable={false}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Bing",
                      "value": "Bing",
                    },
                    Object {
                      "label": "Crackle",
                      "value": "Crackle",
                    },
                    Object {
                      "label": "Down",
                      "value": "Down",
                    },
                    Object {
                      "label": "Hello",
                      "value": "Hello",
                    },
                    Object {
                      "label": "Ripple",
                      "value": "Ripple",
                    },
                    Object {
                      "label": "Upstairs",
                      "value": "Upstairs",
                    },
                  ]
                }
                value={
                  Object {
                    "label": "Bing",
                    "value": "Bing",
                  }
                }
              />
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Mattermost Desktop Apps."
                id="channel.user.settings.notifications.sounds_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`components/channel_notifications_modal/DesktopNotificationSection should show reset to defaults message 1`] = `
<SettingItemMax
  actionButton={
    <button
      aria-labelledby="desktopTitle desktopAction"
      className="color--link cursor--pointer style--none text-left"
      id="desktopAction"
      onClick={[Function]}
    >
      <RepeatIcon />
      <FormattedMessage
        defaultMessage="Reset to defaults"
        id="channel.user.settings.reset_defaults"
      />
    </button>
  }
  containerStyle=""
  extraInfo={
    <ExtraInfo
      section="desktop"
    />
  }
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <React.Fragment>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send desktop notifications"
              id="channel.user.settings.notifications.desktop"
            />
          </legend>
          <fieldset>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelNotificationAllActivity"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                <Describe
                  memberNotifyLevel="all"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={true}
                  id="channelNotificationMentions"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="mention"
                />
                <Describe
                  memberNotifyLevel="mention"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelNotificationNever"
                  name="channelDesktopNotifications"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <Describe
                  memberNotifyLevel="none"
                  section="desktop"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Desktop notifications are available on Firefox, Safari and Chrome."
                id="channel.user.settings.notifications.info"
              />
            </div>
          </fieldset>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <FormattedMessage
                defaultMessage="Notification sound"
                id="channel.user.settings.notifications.desktop.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={true}
                  id="soundOn"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="On"
                  id="channel.user.settings.notifications.on"
                />
              </label>
              <br />
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="soundOff"
                  name="notificationSounds"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="Off"
                  id="channel.user.settings.notifications.off"
                />
              </label>
              <br />
            </div>
            <div
              className="pt-2"
            >
              <StateManager
                className="react-select notification-sound-dropdown"
                classNamePrefix="react-select"
                clearable={false}
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                id="displaySoundNotification"
                isSearchable={false}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Bing",
                      "value": "Bing",
                    },
                    Object {
                      "label": "Crackle",
                      "value": "Crackle",
                    },
                    Object {
                      "label": "Down",
                      "value": "Down",
                    },
                    Object {
                      "label": "Hello",
                      "value": "Hello",
                    },
                    Object {
                      "label": "Ripple",
                      "value": "Ripple",
                    },
                    Object {
                      "label": "Upstairs",
                      "value": "Upstairs",
                    },
                  ]
                }
                value={
                  Object {
                    "label": "Bing",
                    "value": "Bing",
                  }
                }
              />
            </div>
            <div
              className="mt-5"
            >
              <FormattedMessage
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Mattermost Desktop Apps."
                id="channel.user.settings.notifications.sounds_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  server_error=""
  submit={[Function]}
  title={
    <SectionTitle
      section="desktop"
    />
  }
  updateSection={[Function]}
/>
`;
